src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java

```diff
@@ -862,6 +862,8 @@ public boolean apply(@Nullable FlowNode input) {
     @Issue("JENKINS-36547")
     @Test public void reuseNodesWithSameLabelsInDifferentReorderedStages() {
         story.then(r -> {
+            // Note: for Jenkins versions > 2.65, the number of agents must be increased to 5.
+            // This is due to changes in the Load Balancer (See JENKINS-60563).
             for (int i = 0; i < 3; ++i) {
```

Comment:
You can do something like this so the test passes against all versions:

```suggestion
            // Note: for Jenkins versions > 2.265, the number of agents must be increased to 5.
            // This is due to changes in the Load Balancer (See JENKINS-60563).
            int totalAgents = Jenkins.getVersion().isNewerThan(new VersionNumber("2.265")) ? 5 : 3;
            for (int i = 0; i < totalAgents; ++i) {
```

You would also need to add an import for `hudson.util.VersionNumber` and I would update the Jenkinsfile to test against both the current baseline `jenkins: null` and the LTS candidate (`jenkins: 2.277`, and I would add this one to the `java: 11` branch rather than adding another parallel branch).